### ISSUE TITLE
NU1301 can be a Handshake timeout

### ISSUE THREAD

**Issue by pcbl** [NuGet/docs.microsoft.com-nuget#2930]
By Default, the Handshake with the Nuget server is 5 seconds. Sometimes I might end up getting the NU1301 with a message like this:
```
C:\\Users\\VssAdministrator\\.nuget\\plugins\\netcore\\CredentialProvider.Microsoft\\CredentialProvider.Microsoft.dll'. Plugin 'CredentialProvider.Microsoft' failed within 5.099 seconds with exit code -1. 
```

followed by
```
error NU1301: Unable to load the service index for source https://MYSERVER/nuget/v3/index.json.
```

By simply seeting the environment variable to for example 30 seconds solved my issue!

```
  - name: NUGET.PLUGIN.HANDSHAKE.TIMEOUT.IN.SECONDS
    value: 30 
  - name: NUGET.PLUGIN.REQUEST.TIMEOUT.IN.SECONDS
    value: 30    
```
This article describes it on a good way:
https://ryanhayes.net/azure-devproblem-starting-the-plugin-credentialprovider-microsoft-solved/

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: ddfddc4f-d20c-a078-4e72-a997633a48d9
* Version Independent ID: ddfddc4f-d20c-a078-4e72-a997633a48d9
* Content: [NuGet Error NU1301](https://learn.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1301)
* Content Source: [docs/reference/errors-and-warnings/NU1301.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/main/docs/reference/errors-and-warnings/NU1301.md)
* Product: **nuget**
* GitHub Login: @nkolev92
* Microsoft Alias: **jodou**

**Comment by nkolev92** [comment]
Hey @pcbl good news is that we've changed the default for the plugin timeout to 30 already, so hopefully you see less of the 1301 with the default configuration.

**Comment by nkolev92** [comment]
Closing as NU1301 is the bucket for all inaccessible source issues.

**Comment by MisinformedDNA** [comment]
@nkolev92 Where was this change made? I'm still seeing timeouts at less than 6 seconds.

> ##[error]C:\Program Files\dotnet\sdk\6.0.406\NuGet.targets(132,5): Error : Problem starting the plugin 'C:\Users\VssAdministrator\.nuget\plugins\netcore\CredentialProvider.Microsoft\CredentialProvider.Microsoft.dll'. Plugin 'CredentialProvider.Microsoft' failed within 5.795 seconds with exit code -1.

**Comment by nkolev92** [comment]
The issue is https://github.com/NuGet/Home/issues/11793, merged in 6.4 of NuGet. 
The change is in 7.0.100 SDK and later. https://learn.microsoft.com/en-us/nuget/release-notes/nuget-6.4
